addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3200,6 +3217,23 @@ export class OdooEditor extends EventTarget {
                 list.replaceChildren(...rangeContent.childNodes);
                 rangeContent = list;
             }
+            if (rangeContent.firstChild.nodeName === 'TR') {
+                // Clone the whole table rather than just its rows. If several
+                // tables are selected we don't have this problem and if not all
+                // rows are selected we enter the `hasPartiallySelectedTable`
+                // case.
```

Comment:
So... from this comment, I understand that there might be in fact another simpler way to check if a table is partially selected that the complicated code above ? Something like checking if there are still other TR before the rangeContent.firstChild or after the rangeContent.lastChild ? Otherwise I probably didn't understand the comment properly. I feel like the table cases should all be handled here rather than one case in a if above and the other ones here. Let me know what you think or if I misunderstood something.

Reply:
With `rangeContent` that wouldn't work because it's in a `DocumentFragment` but I could probably do all this here indeed, and look for the first and last `tr` in the selection to do what you suggest. I'll try.